src/test/java/io/axoniq/axonserver/connector/event/impl/BufferedAggregateEventStreamTest.java

```diff
@@ -0,0 +1,42 @@
+package io.axoniq.axonserver.connector.event.impl;
+
+import io.axoniq.axonserver.connector.impl.StreamClosedException;
+import io.axoniq.axonserver.grpc.event.Event;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+class BufferedAggregateEventStreamTest {
+
+    private BufferedAggregateEventStream testSubject;
+
+    @BeforeEach
+    void setUp() {
+        testSubject = new BufferedAggregateEventStream(10);
+    }
+
+    @Test
+    void testEventStreamPropagatesErrorOnHasNext() {
+        testSubject.onError(new RuntimeException("Mock"));
+
+        Assertions.assertThrows(StreamClosedException.class,
```

Comment:
Super nit: I'd use a static import here, to be inline with the other `Assertion` usages like `assertTrue` or `assertEquals`